tests/operators/test_python.py

```diff
@@ -327,6 +328,98 @@ def test_python_operator_python_callable_is_callable(self):
+            return {"x": x, "y": y}
+
+        assert identity_dict(5, 5).operator.multiple_outputs is True   # pylint: disable=maybe-no-member
+
+        @task_decorator
+        def identity_tuple(x: int, y: int) -> Tuple[int, int]:
+            return x, y
+
+        assert identity_tuple(5, 5).operator.multiple_outputs is True  # pylint: disable=maybe-no-member
+
+        @task_decorator
+        def identity_int(x: int) -> int:
+            return x
+
+        assert identity_int(5).operator.multiple_outputs is False   # pylint: disable=maybe-no-member
+
+        @task_decorator
+        def identity_notyping(x: int):
+            return x
+
+        assert identity_notyping(5).operator.multiple_outputs is False   # pylint: disable=maybe-no-member
+
+    @conf_vars({("core", "enable_xcom_pickling"): "True"})
+    def test_multiple_outputs_set(self):
+        @task_decorator
+        def identity(x: int, y: int) -> Set[int]:
+            return {x, y}
+
+        with self.dag:
+            res = identity(5, 6)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        res.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)  # pylint: disable=maybe-no-member
+
+        ti = dr.get_task_instances()[0]
+
+        assert ti.xcom_pull() == {5, 6}  # pylint: disable=maybe-no-member
+
+    def test_multiple_outputs_list(self):
+        @task_decorator
+        def identity(x: int, y: int) -> List[int]:
+            return [x, y]
+
+        with self.dag:
+            res = identity(5, 6)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        res.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)  # pylint: disable=maybe-no-member
+
+        ti = dr.get_task_instances()[0]
+
+        assert ti.xcom_pull() == [5, 6]  # pylint: disable=maybe-no-member
+        assert ti.xcom_pull(key="return_value_0") == 5
+        assert ti.xcom_pull(key="return_value_1") == 6
+
+    def test_multiple_outputs_tuple(self):
+        @task_decorator
+        def identity_tuple(x: int, y: int) -> Tuple[int, int]:
+            return x, y
+
+        with self.dag:
+            ident = identity_tuple(5, 6)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        ident.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)  # pylint: disable=maybe-no-member
+
+        ti = dr.get_task_instances()[0]
+
+        assert ti.xcom_pull(key="return_value_0") == 5
+        assert ti.xcom_pull(key="return_value_1") == 6
+
```

Comment:
What happens if I manually set `multiple_outputs=False` but it has annotations?

Reply:
Hmm ohh good q, right now it would override but handling that would make this more complex too, like if we set it to none by default and only infer from typings if it's None,

Reply:
Yup, I would vote in favour of using the None strategy you just mentioned

Reply:
Let's add a test regarding this new case where is set to false and annotations are set.